platforms/ios/src/TangramMap/TGSceneUpdateStatus.h

```diff
@@ -0,0 +1,38 @@
+//
+//  TGSceneUpdateStatus.h
+//  TangramMap
+//
+//  Created by Karim Naaji on 3/07/17.
+//  Copyright (c) 2017 Mapzen. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@class TGSceneUpdate;
+
+/// Description of error status triggered by a scene update
+typedef NS_ENUM(NSInteger, TGSceneUpdateError) {
+    /// The path of the scene update was not found on the scene file
+    TGSceneUpdateErrorPathNotFound,
+    /// The YAML syntax of the scene udpate path has a syntax error
+    TGSceneUpdateErrorPathYAMLSyntaxError,
+    /// The YAML syntax of the scene update value has a syntax error
+    TGSceneUpdateErrorValueYamlSyntaxError,
+};
+
+/**
+ Holds information relative to a scene update error status
+ */
+@interface TGSceneUpdateStatus : NSObject
+
+/// A scene update holding the path and the scene update value
+@property (readonly, nonatomic) TGSceneUpdate* sceneUpdate;
+
+/// The type of error the scene update triggered when applied to the scene
+@property (readonly, nonatomic) TGSceneUpdateError error;
```

Comment:
This is a good start. Where we could go with this would be to use TGSceneUpdateError types as the error codes for NSError.

I'd imagine, from an implementing developer side, a nullable property (since NSError is an object) that you'd check to see if its nil or not, and then check the code.

So it build an NSError at bare minimum you need an Error domain (a string constant) and an error code. Then in the implementation it's as simple as constructing it and passing it here as a property instead of the enum.